### DISCUSSION TITLE
LuaNLP Feature Requests

### DISCUSSION THREAD

**Discussion by pncnmnp** [pncnmnp/LuaNLP#1] [Ideas]
At present, this library supports a handful of algorithms. Also, I am in the process of adding support for **Named entity recognition** and **Word sense disambiguation**. If there are any specific algorithms you would like me to port to Lua, add them below. 

I will try implementing 1 feature every month.

**NOTE:** If one of your feature requests has already been mentioned below, add a :+1: instead.

**Comment by pncnmnp** [comment]
**UPDATE: (13/3/21)** Averaged Perceptron based Named-entity Recognition has been added.

**Comment by kennethcochran** [comment]
Does LuaNLP have any support for NLG?  I read through the readme and though much of it was over my head I didn't see anything specific to language generation.

I have a project I'm working on and I'm toying with attempting to use NLG to produce natural language descriptions from a structured event log.

**Reply by pncnmnp** [reply]
Hi Kenneth,
LuaNLP does not have support for NLG.

I am curious, what kind of NLG support are you looking for?
Is it similar to [a context-free grammar based approach](https://nlgen.readthedocs.io/en/latest/cfg_tutorial.html)?
